axiom/nr_segment_message.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_axiom.h"
+
+#include <stdio.h>
+
+#include "nr_header.h"
+#include "nr_segment_message.h"
+#include "nr_segment_private.h"
+#include "util_strings.h"
+#include "util_url.h"
+
+/*
+ * Purpose : Set all the typed message attributes on the segment.
+ */
+static void nr_segment_message_set_attrs(
+    nr_segment_t* segment,
+    const nr_segment_message_params_t* params,
+    nrtxnopt_t options) {
+  nr_segment_message_t message_attributes = {0};
+
+  message_attributes.message_action = params->message_action;
```

Comment:
Does this function need a `NULL` check for `params`?

Reply:
Technically yes, but it is a private function to this file and the calling function already checked params for NULL so it seemed not needed.
But if you feel strongly about this, happy to put it in.

Reply:
This is a recurring topic that comes up and I don't have a feel for how we've decided to handle it.  Maybe having a naming convention for functions to indicate they are not robustly checking arguments would be a long term solution.

Reply:
Updated comment here: [97218d6](https://github.com/newrelic/newrelic-php-agent/pull/991/commits/97218d6120f6039daa904dcff59bcc18d54c45e1)